src/main/java/uk/gov/companieshouse/companyprofile/delta/mapper/CompanyProfileEnumMapper.java

```diff
@@ -0,0 +1,107 @@
+package uk.gov.companieshouse.companyprofile.delta.mapper;
+
+import org.mapstruct.*;
+import uk.gov.companieshouse.api.company.Accounts;
+import uk.gov.companieshouse.api.company.CompanyProfile;
+import uk.gov.companieshouse.api.company.Data;
+import uk.gov.companieshouse.api.company.LastAccounts;
+import uk.gov.companieshouse.api.delta.BooleanFlag;
+import uk.gov.companieshouse.api.delta.CompanyDelta;
+
+
+import javax.validation.constraints.Null;
+import java.util.HashMap;
+
+@Mapper(componentModel = "spring")
+public abstract class CompanyProfileEnumMapper {
+
+    public abstract CompanyProfile companyDeltaToCompanyProfile(CompanyDelta companyDelta);
+
+    /**maps BooleanFlag to Boolean. */
+    public Boolean flagToBoolean(BooleanFlag flag) {
+        return flag == null ? null : flag.getValue().equals("1");
+    }
+
+    /**Maps enum type to string. */
+    @AfterMapping
+    public void mapEnums(@MappingTarget CompanyProfile target, CompanyDelta source) {
+        Data data = target.getData( );
+        if(data == null) {
+            data = new Data() ;
+        }
+        BooleanFlag subTypeFlag = source.getSubtype( );
+        String subType = null;
+        if(subTypeFlag != null){
+            subType = subTypeFlag.getValue();
+        }
+        HashMap<String,String> subTypeMap = MapperUtils.getSubTypeMap( );
+        data.setSubtype(subTypeMap.getOrDefault(subType,null));
+        target.setData(data);
+    }
+
+    /** Maps enum status to string. */
+    @AfterMapping
+    public void mapEnumsStatus(@MappingTarget CompanyProfile target, CompanyDelta source) {
+        Data data = target.getData();
+        String statusType = source.getStatus();
```

Comment:
The variable name doesn't really make sense to me here, why 'statusType' not just 'status?